public/app/themes/justice/src/patch/js/ccfw-storage.js

```diff
@@ -1,29 +1,82 @@
 /**
  * A script for determining our storage method
  *
- * Falls back to cookies if localStorage is unavailable
+ * Falls back to cookies if localStorage is unavailable, otherwise
+ * we disable storage all together.
  */
-if (window.localStorage === undefined) {
-    Storage = {
-        getItem: (key) => {
-            const value = `; ${document.cookie}`;
-            const parts = value.split(`; ${key}=`);
+let Storage = {
+    disabled: true
+}
+
+const CCFWStorage = {
+    /**
+     * Both local() and cookie() return booleans
+     * false = invalid
+     * true = valid
+     * First we test for localStorage, if this is false, we test for cookies
+     * If cookie returns true, valid() returns true, otherwise, valid() returns
+     * false.
+     *
+     * @returns {boolean}
+     */
+    valid: () => {
+        return CCFWStorage.local() || CCFWStorage.cookie()
+    },
+    local: () => {
+        let mod = 'ccfw';
+        try {
+            localStorage.setItem(mod, mod);
+            localStorage.removeItem(mod);
+            return true;
+        } catch (e) {
+            return false
+        }
+    },
+    cookie: () => {
+        try {
+            // try and set it...
+            document.cookie = 'cookie_test=1'
+            // test it exists...
+            const cookiesEnabled = document.cookie.indexOf('cookie_test=') !== -1
+            // remove it...
+            document.cookie = 'cookie_test=1; expires=Thu, 01-Jan-1970 00:00:01 GMT'
+            return cookiesEnabled
+        } catch (e) {
+            // Catch and handle the error if cookies are disabled.
+            return false
+        }
+    }
+}
 
-            if (parts.length === 2) {
-                let part = parts.pop().split(';').shift();
-                return part === '' ? null : part;
+if (CCFWStorage.valid()) {
```

Comment:
A minor, but could we avoid nested `if`s here?

Reply:
I see your point and value it, however, the `valid()` call here must encompass checks for `local()` and `cookie()`. When we pass valid(), we know that either local() or cookie() is alive, we have to work out which one. If `local()` is not accessible, we know that `cookie()` is.